### PR TITLE
ns2 pooladmin swap spell

### PR DESCRIPTION
**Pull request by ilinzweilin**

### CODE REVIEW

File: src/addresses_ns2.sol [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	address constant public SHELF = 0x7d057A056939bb96D682336683C10EC89b78D7CE;
+	address constant public COLLECTOR = 0x62f290512c690a817f47D2a4a544A5d48D1408BE;
+	address constant public FEED = 0x41fAD1Eb242De19dA0206B0468763333BB6C2B3D;
+	address constant public JUNIOR_OPERATOR = 0x4c4Cc6a0573db5823ECAA1d1d65EB64E5E0E5F01;
+	address constant public SENIOR_OPERATOR = 0x230f2E19D6c2Dc0c441c2150D4dD9d67B563A60C;
+	address constant public JUNIOR_TOKEN = 0x961e1d4c9A7C0C3e05F17285f5FA34A66b62dBb1;
+	address constant public SENIOR_TOKEN = 0xE4C72b4dE5b0F9ACcEA880Ad0b1F944F85A9dAA0;
+	address constant public JUNIOR_MEMBERLIST = 0x42C2483EEE8c1Fe46C398Ac296C59674F9eb88CD;
+	address constant public SENIOR_MEMBERLIST = 0x5B5CFD6E45F1407ABCb4BFD9947aBea1EA6649dA;
+	address constant public COORDINATOR = 0xcC7AFB5DeED34CF67E72d4C53B142F44c9268ab9;
```
Comment (offerijns) [inline src/addresses_ns2.sol:15]: This one is old, it was swapped again in July 2021.

```suggestion
	address constant public COORDINATOR = 0x22a1caca2EE82e9cE7Ef900FD961891b66deB7cA;
```

File: src/spell.sol [hunk 2]
```diff
@@ -1,101 +1,154 @@
...
+    address public LEVEL1_ADMIN4 = 0xEf270f8877Aa1875fc13e78dcA31f3235210368f;
+    address public LEVEL1_ADMIN5 = 0xddEa1De10E93c15037E83b8Ab937A46cc76f7009;
+    address public AO_POOL_ADMIN = 0x7Cae9bD865610750a48575aF15CAFe1e460c96a8;
+
+    address public POOL_REGISTRY = 0xddf1C516Cf87126c6c610B52FD8d609E67Fb6033;
 
-    // TODO: replace the following address
-    address constant public RESERVE_NEW = 0x1f5Fa2E665609CE4953C65CE532Ac8B47EC97cD5;
+    // Adam to add correct IPFS hash
+    string constant public IPFS_HASH = "QmS3D3EAySc9b2CVrNnPV2ueo6tBvTFi49kewttQq4vQLw";
```
Comment (offerijns) [inline src/spell.sol:80]: Leaving a comment here so we don't forget to update this

File: src/spell.sol [hunk 3]
```diff
@@ -1,101 +1,154 @@
...
+        AuthLike(POOL_ADMIN).rely(LEVEL3_ADMIN1);
+        // set lvl1 admins
+        PoolAdminLike(POOL_ADMIN).setAdminLevel(LEVEL1_ADMIN1, 1);
+        PoolAdminLike(POOL_ADMIN).setAdminLevel(LEVEL1_ADMIN2, 1);
+        PoolAdminLike(POOL_ADMIN).setAdminLevel(LEVEL1_ADMIN3, 1);
+        PoolAdminLike(POOL_ADMIN).setAdminLevel(LEVEL1_ADMIN4, 1);
+        PoolAdminLike(POOL_ADMIN).setAdminLevel(LEVEL1_ADMIN5, 1);
+        PoolAdminLike(POOL_ADMIN).setAdminLevel(AO_POOL_ADMIN, 1);
+        AuthLike(JUNIOR_MEMBERLIST).rely(MEMBER_ADMIN);
+        AuthLike(SENIOR_MEMBERLIST).rely(MEMBER_ADMIN);
```
Comment (offerijns) [inline src/spell.sol:147]: Afaik these were already done in earlier spells.
```suggestion
```

**Review by offerijns — APPROVED** [review]

File: src/addresses_ns2.sol (hunk 1 above)
Comment (ilinzweilin) [reply src/addresses_ns2.sol:15]: good catch

**Pull request merged**